lib/SILOptimizer/Utils/Generics.cpp

```diff
@@ -1937,6 +1942,15 @@ static void prepareCallArguments(ApplySite AI, SILBuilder &Builder,
   }
 }
 
+static void cleanupCallArguments(SILBuilder &builder, SILLocation loc,
+                                 ArrayRef<SILValue> values) {
+  for (SILValue v : values) {
+    if (auto *lbi = dyn_cast<LoadBorrowInst>(v)) {
+      builder.createEndBorrow(loc, lbi);
```

Comment:
I am a little nervous about this. I think I am going to change values here to have a bit that says it needs an end borrow or something like that.

Reply:
You are right: what if there was a load_borrow before specialization? The you would insert a wrong end_borrow